pytorch_lightning/utilities/debugging.py

```diff
@@ -16,39 +16,45 @@
 import time
 from collections import Counter
 from functools import wraps
-from typing import Any, Callable, Optional
+from typing import Any, Callable, Dict, List, Optional, Union
 
+import torch
+from torch.utils.data import DataLoader
 
-def enabled_only(fn: Callable):
+import pytorch_lightning as pl
+
+
+def enabled_only(fn: Callable) -> Optional[Callable]:
     """Decorate a logger method to run it only on the process with rank 0.
 
     Args:
         fn: Function to decorate
     """
 
     @wraps(fn)
-    def wrapped_fn(self, *args, **kwargs):
+    def wrapped_fn(self: Callable, *args: Any, **kwargs: Any) -> Optional[Any]:
         if self.enabled:
             fn(self, *args, **kwargs)
+        return None
 
     return wrapped_fn
 
 
 class InternalDebugger:
-    def __init__(self, trainer):
+    def __init__(self, trainer: "pl.Trainer") -> None:
         self.enabled = os.environ.get("PL_DEV_DEBUG", "0") == "1"
         self.trainer = trainer
-        self.saved_train_losses = []
-        self.saved_val_losses = []
-        self.saved_test_losses = []
-        self.early_stopping_history = []
-        self.checkpoint_callback_history = []
-        self.events = []
-        self.saved_lr_scheduler_updates = []
-        self.train_dataloader_calls = []
-        self.val_dataloader_calls = []
-        self.test_dataloader_calls = []
-        self.dataloader_sequence_calls = []
+        self.saved_train_losses: List[Dict[str, Union[float, torch.Tensor, object]]] = []
```

Comment:
what's the object type for? would it not be typed as Any?

Reply:
Yes, it seems `Any` typing is appropriate for this.